### PR TITLE
Improve compiler error message/153

### PR DESCRIPTION
**Pull request by LukaszKontowski**
1. Added helpful error message for compilation failures caused by using JDK version lower than 11.
2. Updated LICENSE file with 2022 year.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -21,6 +21,8 @@ sonatypeProfileName := "org.virtuslab"
 
 Global / onChangedBuildSource := ReloadOnSourceChanges
 
+(Compile / compile) := (Compile / compile).dependsOn(checkCurrentJavaVersion).value
```
Comment (PawelLipski) [inline build.sbt:24]: Actually.... would it make sense to put the check into `initialize ~= { _ => ... }` block instead? 🤔

File: build.sbt [hunk 2]
```diff
@@ -202,3 +204,13 @@ lazy val dumpPersistenceSchemaCompilerPlugin = (projectMatrix in file("dump-pers
     assembly / assemblyJarName := s"${name.value}_${scalaBinaryVersion.value}-${version.value}.jar", //Warning: this is a default name for packageBin artefact. Without explicit rename of packageBin will result in race condition
     addArtifact(Compile / assembly / artifact, assembly))
   .jvmPlatform(scalaVersions = supportedScalaVersions)
+
+lazy val checkCurrentJavaVersion =
+  taskKey[Unit]("checks whether SBT has been started with proper java version (at least 11)")
+
+checkCurrentJavaVersion := {
+  if (sys.props.getOrElse("java.specification.version", "0").toDouble < 11.0) {
```
Comment (PawelLipski) [inline build.sbt:212]: Does the project compile with no issues on Java 14/17?

Comment (LukaszKontowski) [reply build.sbt:212]: Yes, it compiles with no issues with both 14 and 17 - so in my opinion it is ok to say that JDK version should be at least 11 (might be higher)

File: build.sbt (hunk 1 above)
Comment (LukaszKontowski) [reply build.sbt]: I replaced this Compile / compile line with initialize block - please take a look

**Review by PawelLipski — APPROVED** [review]
:fantastycznie:

**Comment by PawelLipski** [comment]
Btw as you can see, true merges (aka 2-parent merges) are NOT enabled in this repo... see https://slides.com/plipski/git-machete/#/8 and the following slides for rationale

**Pull request merged**